multi_host_migration/tests/migration_multi_host_auto_converge.py

```diff
@@ -0,0 +1,498 @@
+                    break
+            logging.info("cpu average list: %s" % cpu_average)
+            logging.info("memory average list: %s" % memory_average)
+            sar_output.append(cpu_average)
+            sar_output.append(memory_average)
+
+        @error.context_aware
+        def check_sar_output(self):
+
+            """
+            Compare the output for (1) default auto-converge setting (off)
+            and (2) auto-converge on, the guest performance should not be
+            effected obviously with auto-converge on. (30% is acceptance)
+            """
+
+            logging.info("The sar output list: %s" % sar_output)
+            cpu_average = zip(sar_output[0], sar_output[2])
+            memory_average = zip(sar_output[1], sar_output[3])
+            for i in cpu_average:
+                if abs(i[0] - i[1]) > 30:
+                    raise error.TestFail("The guest performance should "
+                                         "not be effected obviously with "
+                                         "auto-converge on.")
+            for i in memory_average:
+                if abs(i[0] - i[1]) > 30:
+                    raise error.TestFail("The guest performance should "
+                                         "not be effected obviously with "
+                                         "auto-converge on.")
+
+        @error.context_aware
+        def get_mig_cpu_throttling_percentage(self, vm):
+
+            """
+            get cpu throttling percentage during migration
+
+            :param vm: vm object
+            """
+
+            error.context("Get cpu throttling percentage during migration",
+                          logging.info)
+            cpu_throttling_percentage = \
+                vm.monitor.info("migrate").get("cpu-throttle-percentage")
+            logging.info("The cpu throttling percentage is %s%%" %
+                         cpu_throttling_percentage)
+            return cpu_throttling_percentage
+
+        @error.context_aware
+        def check_mig_cpu_throttling_percentage(self):
+
+            """
+            check if cpu throttling percentage equal to given value
+            """
+
+            error.context("check cpu throttling percentage during migration",
+                          logging.info)
+            logging.info("The cpu throttling percentage list is %s" %
+                         cpu_throttling_percentage_list)
+            if (self.parameters_value[0] not in cpu_throttling_percentage_list) or \
+                    (sum(self.parameters_value) not in cpu_throttling_percentage_list):
+                raise error.TestFail("The value of cpu throttling percentage "
+                                     "should include: %s %s" %
+                                     (self.parameters_value[0],
+                                      sum(self.parameters_value)))
+            if min(cpu_throttling_percentage_list) != self.parameters_value[0]:
```

Comment:
if cpu_throttling_percentage_list[0] != self.parameters_value[0]:
...
if sum(self.parameters_value) not in cpu_throttling_percentage_list:
...